### PR TITLE
Indirect ConvFit : avoid nullptr if component does not exist

### LINKED ISSUE #25188: Indirect ConvFit GUI crashes with grouped input
See the title.

### PR DESCRIPTION
**Pull request by gvardany**
**Description of work.**
Fixes a hard crash when the analyser is not defined as an instrument component.

**To test:**
Go to Interfaces > Indirect > Data Analysis > ConvFit tab.
Load the attached file in the sample input, wait until it loads.
Mantid should not crash, unlike in current `release-next`.

Chop off `.zip`; it's a nexus file.

[example_red.nxs.zip](https://github.com/mantidproject/mantid/files/2970413/example_red.nxs.zip)

Fixes #25188 . 

*This does not require release notes, since it's a minor change.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/ConvFitModel.cpp [hunk 1]
```diff
@@ -157,9 +157,14 @@ getAnalyser(MatrixWorkspace_sptr workspace) {
         "Could not load instrument resolution from parameter file");
 
   auto component = instrument->getComponentByName(analysers[0]);
-  auto resolutionParameters = component->getNumberParameter("resolution");
-  if (nullptr == component || resolutionParameters.empty())
+  if (component) {
+    auto resolutionParameters = component->getNumberParameter("resolution");
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/ConvFitModel.cpp]: Could `const` be used here?

**Review by robertapplin — CHANGES_REQUESTED** [review]
The code changes look good and ConvFit no longer crashes when loading grouped input. Just a small comment about `const`

File: qt/scientific_interfaces/Indirect/ConvFitModel.cpp (hunk 1 above)
Comment (gvardany) [reply qt/scientific_interfaces/Indirect/ConvFitModel.cpp]: Yes, actually one should also check if the parameter exists before getting it. Do you mind committing it?

**Comment by AnthonyLim23** [comment]
@robertapplin is this ok to go?

**Review by robertapplin — APPROVED** [review]

**Pull request merged**